manager/controllers/utils/unstructured.go

```diff
@@ -47,6 +48,15 @@ func (c UnstructuredAsLabels) Get(label string) string {
 	obj := c.Data.UnstructuredContent()
 	fields := strings.Split(label, ".")
 	// not checking whether the label exists and is valid. Assuming Get is called after Has in labels package evaluation
-	val, _, _ := unstructured.NestedString(obj, fields...)
-	return val
+	// fetch a string value
+	if valStr, _, err := unstructured.NestedString(obj, fields...); err == nil {
+		return valStr
+	}
+	// convert a received interface into a string
+	val, _, _ := unstructured.NestedFieldNoCopy(obj, fields...)
```

Comment:
Is it really okay to ignore the error here? (last `_`)

Reply:
It's supposed to be applied after a successful Has function that checks the value, but I'll add the sanity check.